addons/account/models/account_account.py

```diff
@@ -465,13 +558,57 @@ def _onchange_user_type_id(self):
         elif self.internal_group == 'expense' and not self.tax_ids:
             self.tax_ids = self.company_id.account_purchase_tax_id
 
+    @api.onchange('code')
+    def _onchange_code(self):
+        code, name = self._split_code_name(self.code)
+        if name:
+            self.name = name
+            self.code = code
+
+    @api.onchange('name')
+    def _onchange_name(self):
+        code, name = self._split_code_name(self.name)
+        if code:
+            self.name = name
+            self.code = code
+
     def name_get(self):
         result = []
         for account in self:
-            name = account.code + ' ' + account.name
+            name = account.code + ' ' + (account.name or '')
             result.append((account.id, name))
         return result
 
+    @api.model
+    def name_create(self, name):
+        # Enforce no_quick_create from forms - however name_create is possible when importing from a related model
+        if 'import_file' in self.env.context:
+            code, name = self._split_code_name(name)
+            return self.create({'code': code, 'name': name}).name_get()[0]
+        raise ValidationError(_(
+            "The operation cannot be completed:\n"
+            "- Create/update: a mandatory field is not set.\n"
+            "- Delete: another model requires the record being deleted."
+            " If possible, archive it instead.\n\n"
+            "Model: %(model_name)s (%(model_tech_name)s)\n"
+            "Field: %(field_name)s (%(field_tech_name)s)\n",
+            model_name=self._description,
+            model_tech_name=self._name,
+            field_name='Type',
+            field_tech_name='user_type_id',
+        ))
+
+    def _split_code_name(self, code_name):
+        results = ACCOUNT_REGEX.match(code_name or '').groups('')
```

Comment:
Parameter to `groups` is useless (it's the same as leaving it out), and call to `groups` is useless, match objects have been indexable directly since Python 3.6

Reply:
updated